firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/SPI/BMP388.h

```
@@ -0,0 +1,249 @@
+                            bool pressEnable,
+                            bool timeEnable,
+                            bool stopOnFull);
+  /**
+   * @brief  Get the FIFO length
+   * @param  address of FIFO length variable
+   * @return SPIBMP388Status returns OK/writeError
+   */
+  SPIBMP388Status getFIFOLength();
+
+  /**
+   * @brief  To get the compensate temperature data
+   * @param  compensateTemperature updated with the compensate temperature
+   * @return SPIBMP388Status ok/readError
+   */
+  SPIBMP388Status readCompensateTemperature(float &compensateTemperature);
+
+  /**
+   * @brief  To get the compensate pressure data
+   * @param  compensatePressure updated with the compensate pressure
+   * @return SPIBMP388Status ok/readError
+   */
+  SPIBMP388Status readCompensatePressure(float &compensatePressure);
+
+ private:
+
+  /**
+  * @ Quantized Trim Variables
+  */
+  struct bmp388QuantizedCalibData {
+    double par_t1;
+    double par_t2;
+    double par_t3;
+    double par_p1;
+    double par_p2;
+    double par_p3;
+    double par_p4;
+    double par_p5;
+    double par_p6;
+    double par_p7;
+    double par_p8;
+    double par_p9;
+    double par_p10;
+    double par_p11;
+    double t_lin;
+  };
+  /**
+   * @brief  Writes the data into the register
+   * @param  buff buffer data to write
+   * @param  count size of buffer
+   * @return SPIBMP388Status returns ok/writeError
+   */
+  SPIBMP388Status write(uint8_t *buff, size_t count);
+
+  /**
+   * @brief  Writes and read the data from the register
+   * @param  txbuf data to write into DI pin
+   * @param  rxbuf data read from DO pin
+   * @return SPIBMP388Status returns ok/readError
+   */
+  SPIBMP388Status writeRead(uint8_t *txbuf, uint8_t *rxbuf, size_t count);
+
+  /**
+   * @brief  reads the CalibrationData data from NVM
+   * @param  QuantizedCalibData updates the structure bmp388QuantizedCalibData
+   * @return None
+   */
+  void ReadCalibrationData();
```

Is it possible for this operations to fail? If so, do we need to return an error code or status code?